operator/controllers/instance_controller.go

```diff
@@ -0,0 +1,461 @@
+		if err != nil {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values.
+			return ctrl.Result{}, nil
+		}
+		if !result.Valid() {
+			log.Error(err, "failed to validate values")
+			// TODO: update status with invalid values. Including specific errors returned from
+			// result.Errors().
+			return ctrl.Result{}, nil
+		}
+	}
+
+	valuesSecretName := fmt.Sprintf("metabroker-%s-values", instance.Name)
+	if created, err := r.valuesSecret(ctx, instance, valuesSecretName, instance.Namespace); err != nil {
+		return ctrl.Result{}, err
+	} else if created {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	helmInstanceName := fmt.Sprintf("metabroker-%s", instance.Name)
+	podName := fmt.Sprintf("metabroker-%s-provision", instance.Name)
+	return r.provisioningPod(ctx, instance, plan, helmInstanceName, podName, instance.Namespace, valuesSecretName)
+}
+
+func (r *InstanceReconciler) setOwnership(
+	ctx context.Context,
+	instance *servicebrokerv1alpha1.Instance,
+	plan *servicebrokerv1alpha1.Plan,
+) error {
+	if err := ctrl.SetControllerReference(plan, instance, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set ownership: %w", err)
+	}
+	return nil
+}
+
+// valuesSecret creates a Secret containing the values.yaml content if it doesn't exist yet.
+// The first return parameter represents whether the Secret was created or not.
+func (r *InstanceReconciler) valuesSecret(
+	ctx context.Context,
+	instance *servicebrokerv1alpha1.Instance,
+	valuesSecretName string,
+	namespace string,
+) (bool, error) {
+	namespacedName := types.NamespacedName{
+		Name:      valuesSecretName,
+		Namespace: namespace,
+	}
+	current := &corev1.Secret{}
+	if err := r.Get(ctx, namespacedName, current); err != nil {
+		if !errors.IsNotFound(err) {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		desired := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespacedName.Name,
+				Namespace: namespacedName.Namespace,
+				// TODO: add proper labels.
+			},
+			Data: map[string][]byte{"values.yaml": []byte(instance.Spec.Values)},
+		}
+		if err := ctrl.SetControllerReference(instance, desired, r.Scheme); err != nil {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		if err := r.Create(ctx, desired); err != nil {
+			return false, fmt.Errorf("failed to process values secret: %w", err)
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *InstanceReconciler) provisioningPod(
+	ctx context.Context,
+	instance *servicebrokerv1alpha1.Instance,
+	plan *servicebrokerv1alpha1.Plan,
+	helmInstanceName string,
+	podName string,
+	namespace string,
+	valuesSecretName string,
+) (ctrl.Result, error) {
+	namespacedName := types.NamespacedName{
+		Name:      podName,
+		Namespace: namespace,
+	}
+
+	currentServiceAccount := &corev1.ServiceAccount{}
+	if err := r.Get(ctx, namespacedName, currentServiceAccount); err != nil {
```

Comment:
This only ensures that the service account exists, not that it's in the desired state, right? (Same with all the other resources below.)

Reply:
Yes, we let k8s do its job. It won't spawn the pod while things are not correctly set yet.

Reply:
This specific case was already addressed in b08a3fed with the `Update` instructions.